src/core/unit.h

```diff
@@ -300,6 +300,11 @@ typedef struct Unit {
         Set *ip_bpf_custom_egress;
         Set *ip_bpf_custom_egress_installed;
 
+        /* BPF programs managed (e.g. loaded to kernel) by an entity external to systemd,
+         * attached to unit cgroup by provided program fd and attach type. */
+        Hashmap *bpf_foreign_next_by_key;
```

Comment:
hmm, why "next" in the name?

Reply:
"next" is where loaded programs were stored prior to being put to "installed" set and attached.
Since "foreign" programs attachments are not preserved through daemon-reload,reexec, simplified that part.